saltproc/serpent_depcode.py

```diff
@@ -352,63 +340,61 @@ def read_plaintext_file(self, file_path):
-        template_data,
-        reactor,
-            current_depstep_idx):
-        """Adds or replaces the ``set power P dep daystep DEPSTEP`` line in
-        the Serpent2 input file. This line defines depletion history and power
-        levels with respect to the depletion step in the single run and
-        activates depletion calculation mode.
+            file_lines = file.readlines()
+        return file_lines
+
+    def set_power_load(self,
+                       file_lines,
+                       reactor,
+                       step_idx):
+        """Add power load attributes in a :class:`Reactor` object to the
+        ``set power P dep daystep DEPSTEP`` line in the Serpent2  runtime input
+        file.
 
         Parameters
         ----------
-        template_data : list
-            List of strings parsed from user template file.
+        file_lines : list of str
+            Serpent2 runtime input file.
         reactor : Reactor
             Contains information about power load curve and cumulative
             depletion time for the integration test.
-        current_depstep_idx : int
+        step_idx : int
             Current depletion step.
 
         Returns
         -------
-        template_data : list
-            List of strings containing modified in this function template file.
+        file_lines : list of str
+            Serpent2 runtime input file with power load specification.
 
         """
 
         line_idx = 8  # burnup setting line index by default
-        current_depstep_power = reactor.power_levels[current_depstep_idx]
-        if current_depstep_idx == 0:
-            current_depstep = reactor.dep_step_length_cumulative[0]
+        current_power = reactor.power_levels[step_idx]
+        if step_idx == 0:
+            step_length = reactor.dep_step_length_cumulative[0]
         else:
-            current_depstep = \
-                reactor.dep_step_length_cumulative[current_depstep_idx] - \
-                reactor.dep_step_length_cumulative[current_depstep_idx - 1]
-        for line in template_data:
+            step_length = \
+                reactor.dep_step_length_cumulative[step_idx] - \
+                reactor.dep_step_length_cumulative[step_idx - 1]
+        for line in file_lines:
             if line.startswith('set    power   '):
-                line_idx = template_data.index(line)
-                del template_data[line_idx]
+                line_idx = file_lines.index(line)
+                del file_lines[line_idx]
 
-        template_data.insert(line_idx,  # Insert on 9th line
-                             'set    power   %5.9E   dep daystep   %7.5E\n' %
-                             (current_depstep_power,
-                              current_depstep))
-        return template_data
+        file_lines.insert(line_idx,  # Insert on 9th line
+                          'set    power   %5.9E   dep daystep   %7.5E\n' %
+                          (current_power, step_length))
```

Comment:
Are these values hardcoded into the file lines? Is there any reason someone would need to use different values?

Reply:
The `set power` card is the only way to specify depletion steps in Serpent to my knowledge. Right now, we can only use the `daystep` option, but in #177 I fix this to enable use of every potential time unit a user might want to use.

Reply:
That makes sense, but what about the 5.9E and 7.5E values in there? Are those values, or a way to read in a variable value?

Reply:
I agree

Reply:
This is string formatting. The `5` in `5.9E` indicates we want to display a maximum of 5 digits before the decimal, the `9` indicates we want to display 9 digits after the decimal, and the `E` indicates we display the number in scientific notation:
```python
>>> '%5.9f' % (2.0e-2)
'0.020000000'
>>> '%5.9E' % (2.0e-2)
'2.000000000E-02'
```